conans/model/build_info.py

```diff
@@ -41,50 +47,134 @@ def __init__(self):
         self.description = None  # Description of the conan package
         # When package is editable, filter_empty=False, so empty dirs are maintained
         self.filter_empty = True
+        self._components = OrderedDict()
+
+    @property
+    def _sorted_components(self):
+        """
+        Sorted components from less dependent to the most one (less items in .deps attribute)
+        :return: ordered list of components
+        """
+        return sorted(self._components.values(), key=lambda component: len(component.deps))
```

Comment:
Is this function the responsable to give the data (libs, includes...) in order? Why the `len(components.deps)` is relevant at all?

In this example, what is the correct order?

```
 self.cpp_info["A"].deps=["B"]
 self.cpp_info["C"].deps=["A"]
 self.cpp_info["B"].deps=[]
```

Reply:
You have to sort the libs staring from the less dependent one. In your example, the order should be: B, A, C

Reply:
And why B, A, C and not B, C, A?:

`len(self.cpp_info["A"].deps) == len(self.cpp_info["C"].deps == 1`